doc/source/development/notebooks/processor_examples/string_splitter.ipynb

```diff
@@ -0,0 +1,269 @@
+   ],
+   "source": [
+    "import sys\n",
+    "sys.path.append(\"../../../../../\")\n",
+    "\n",
+    "from logprep.processor.string_splitter.rule import StringSplitterRule\n",
+    "rules_definitions = [\n",
+    "    {\n",
+    "        \"filter\": \"ip_addresses\",\n",
+    "        \"string_splitter\": {\n",
+    "            \"source_fields\": [\"ip_addresses\"],\n",
+    "            \"target_field\": \"ip_addresses\",\n",
+    "            \"overwrite_target\": True\n",
+    "        },\n",
+    "    }\n",
+    "]\n",
+    "rules = [StringSplitterRule._create_from_dict(rule_dict) for rule_dict in rules_definitions]\n",
+    "rules"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "create the processor config:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "processor_config = {\n",
+    "    \"allmighty_string_splitter\": {\n",
+    "        \"type\": \"string_splitter\",\n",
+    "        \"specific_rules\": [\"/dev\"],\n",
+    "        \"generic_rules\": [\"/dev\"],\n",
+    "    }\n",
+    "}\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "create the processor with the factory:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "string_splitter"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from logging import getLogger\n",
+    "from logprep.factory import Factory\n",
+    "\n",
+    "logger = getLogger()\n",
+    "\n",
+    "processor = Factory.create(processor_config, logger)\n",
+    "processor\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "load rules to processor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[filter=\"ip_addresses\", StringSplitterRule.Config(description='', regex_fields=[], tests=[], tag_on_failure=['_string_splitter_failure'], source_fields=['ip_addresses'], target_field='ip_addresses', delete_source_fields=False, overwrite_target=True, extend_target_list=False, delimeter=' ')]"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "for rule in rules:\n",
+    "    processor._specific_tree.add_rule(rule)\n",
+    "    \n",
+    "processor._specific_rules"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Process event"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from copy import deepcopy\n",
+    "\n",
+    "mydocument = deepcopy(document)\n",
+    "processor.process(mydocument)\n"
```

Comment:
I find this example to be confusing. For me, it would be a lot clearer if the source and target field weren't identical.